src/network.py

```diff
@@ -149,33 +192,34 @@ def infer_sequence(self, hidden_state, enc_output, enc_ctxt_mask, max_len=1000):
         batch_size = enc_output.shape[0]
         outputs = []
         stops = []
-        stop_lens = torch.zeros(batch_size).to(enc_output.device)
+        stop_lens = torch.zeros(batch_size, device=enc_output.device)
         
         # get a all 0 frame for first timestep
         input_ = torch.zeros((batch_size, 1, self.postnet.num_mels), device=enc_output.device)
         i = 0
-        keep_gen = torch.all(stop_lens.neq(0)) and i < max_len
+        keep_gen = torch.any(stop_lens.eq(0)) and i < max_len
 
         while keep_gen:
-            dec_out, stop_pred, hidden_state = self.decode(input_, hidden_state, enc_output, enc_ctxt_mask)
+            (dec_out, stop_pred), hidden_state = self.decode(input_, hidden_state, enc_output, enc_ctxt_mask)
+            stop_pred = stop_pred.squeeze()
             stops.append(stop_pred)
             # set stop_lens here!
             outputs.append(dec_out)
-            input_ = outputs[-1]
+            input_ = outputs[-1].detach().permute(0, 2, 1)
```

Comment:
If I'm understanding this right, are we removing the gradients on generating the sequence for the cross-modal loss?

That is, if we had the following data flow:
`speech --> seq2seq --> text --> seq2seq --> speech`

We're only propagating the gradients on `text --> seq2seq --> speech` but not for `speech --> seq2seq --> text`?

Reply:
My bad, I just found out that ‘.detach()’ creates a new tensor sharing values instead of modifying the original one and removing it from the graph. So the loss should propagate here since you add the non-detached one to the list.